### PR TITLE
[Examples] Added public folder to with-docker example

### PR DESCRIPTION
**Pull request by tomevans18**
Added the public folder to the with-docker multistage Dockerfile example along with placeholder files. This ensures access to public folder files (e.g. images) within the dockerised applications and allows for the docker build command to work on pulling the current example (without the need to add placeholder files or edit the Dockerfile).

### CODE REVIEW

**Comment by ijjk** [comment]
# Stats from current PR

<details>
<summary><strong>Default Server Mode</strong> (Increase detected ⚠️)</summary>

<br/>

<details>
<summary><strong>General</strong></summary>

|  | zeit/next.js canary  | tomevans18/next.js with-docker-example-public-folder | Change |
| - | - | - | - |
| buildDuration | 13.4s | 13.3s | -65ms |
| nodeModulesSize | 58.7 MB | 58.7 MB | ✓ |

</details>

<details>
<summary><strong>Page Load Tests</strong> Overall increase ✓</summary>

|  | zeit/next.js canary  | tomevans18/next.js with-docker-example-public-folder | Change |
| - | - | - | - |
| / failed reqs | 0 | 0 | ✓ |
| / total time (seconds) | 2.256 | 2.262 | ⚠️ +0.01 |
| / avg req/sec | 1108.23 | 1105.45 | -2.78 |
| /error-in-render failed reqs | 0 | 0 | ✓ |
| /error-in-render total time (seconds) | 1.452 | 1.421 | -0.03 |
| /error-in-render avg req/sec | 1722.11 | 1758.94 | ⚠️ +36.83 |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | zeit/next.js canary  | tomevans18/next.js with-docker-example-public-folder | Change |
| - | - | - | - |
| main-HASH.js gzip | 6.68 kB | 6.68 kB | ✓ |
| webpack-HASH.js gzip | 746 B | 746 B | ✓ |
| de003c3a9d30..e49d.js gzip | 10.4 kB | 10.4 kB | ✓ |
| framework.HASH.js gzip | 39.1 kB | 39.1 kB | ✓ |
| Overall change | 57 kB | 57 kB | ✓ |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons) Modern</strong></summary>

|  | zeit/next.js canary  | tomevans18/next.js with-docker-example-public-folder | Change |
| - | - | - | - |
| main-HASH.module.js gzip | 5.72 kB | 5.72 kB | ✓ |
| webpack-HASH..dule.js gzip | 746 B | 746 B | ✓ |
| de003c3a9d30..dule.js gzip | 6.85 kB | 6.85 kB | ✓ |
| framework.HA..dule.js gzip | 39.1 kB | 39.1 kB | ✓ |
| Overall change | 52.5 kB | 52.5 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | zeit/next.js canary  | tomevans18/next.js with-docker-example-public-folder | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 26.3 kB | 26.3 kB | ✓ |
| Overall change | 26.3 kB | 26.3 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | zeit/next.js canary  | tomevans18/next.js with-docker-example-public-folder | Change |
| - | - | - | - |
| _app.js gzip | 1.26 kB | 1.26 kB | ✓ |
| _error.js gzip | 3.29 kB | 3.29 kB | ✓ |
| hooks.js gzip | 880 B | 880 B | ✓ |
| index.js gzip | 222 B | 222 B | ✓ |
| link.js gzip | 2.06 kB | 2.06 kB | ✓ |
| routerDirect.js gzip | 280 B | 280 B | ✓ |
| withRouter.js gzip | 278 B | 278 B | ✓ |
| Overall change | 8.26 kB | 8.26 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages Modern</strong></summary>

|  | zeit/next.js canary  | tomevans18/next.js with-docker-example-public-folder | Change |
| - | - | - | - |
| _app.module.js gzip | 599 B | 599 B | ✓ |
| _error.module.js gzip | 2.09 kB | 2.09 kB | ✓ |
| hooks.module.js gzip | 383 B | 383 B | ✓ |
| index.module.js gzip | 223 B | 223 B | ✓ |
| link.module.js gzip | 1.52 kB | 1.52 kB | ✓ |
| routerDirect..dule.js gzip | 279 B | 279 B | ✓ |
| withRouter.m..dule.js gzip | 278 B | 278 B | ✓ |
| Overall change | 5.37 kB | 5.37 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | zeit/next.js canary  | tomevans18/next.js with-docker-example-public-folder | Change |
| - | - | - | - |
| _buildManifest.js gzip | 61 B | 61 B | ✓ |
| _buildManife..dule.js gzip | 61 B | 61 B | ✓ |
| Overall change | 122 B | 122 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | zeit/next.js canary  | tomevans18/next.js with-docker-example-public-folder | Change |
| - | - | - | - |
| index.html gzip | 927 B | 927 B | ✓ |
| link.html gzip | 936 B | 936 B | ✓ |
| withRouter.html gzip | 923 B | 923 B | ✓ |
| Overall change | 2.79 kB | 2.79 kB | ✓ |

</details>

</details>
<hr/>
<details>
<summary><strong>Serverless Mode</strong></summary>

<br/>

<details>
<summary><strong>General</strong></summary>

|  | zeit/next.js canary  | tomevans18/next.js with-docker-example-public-folder | Change |
| - | - | - | - |
| buildDuration | 14.1s | 13.8s | -340ms |
| nodeModulesSize | 58.7 MB | 58.7 MB | ✓ |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | zeit/next.js canary  | tomevans18/next.js with-docker-example-public-folder | Change |
| - | - | - | - |
| main-HASH.js gzip | 6.68 kB | 6.68 kB | ✓ |
| webpack-HASH.js gzip | 746 B | 746 B | ✓ |
| de003c3a9d30..e49d.js gzip | 10.4 kB | 10.4 kB | ✓ |
| framework.HASH.js gzip | 39.1 kB | 39.1 kB | ✓ |
| Overall change | 57 kB | 57 kB | ✓ |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons) Modern</strong></summary>

|  | zeit/next.js canary  | tomevans18/next.js with-docker-example-public-folder | Change |
| - | - | - | - |
| main-HASH.module.js gzip | 5.72 kB | 5.72 kB | ✓ |
| webpack-HASH..dule.js gzip | 746 B | 746 B | ✓ |
| de003c3a9d30..dule.js gzip | 6.85 kB | 6.85 kB | ✓ |
| framework.HA..dule.js gzip | 39.1 kB | 39.1 kB | ✓ |
| Overall change | 52.5 kB | 52.5 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | zeit/next.js canary  | tomevans18/next.js with-docker-example-public-folder | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 26.3 kB | 26.3 kB | ✓ |
| Overall change | 26.3 kB | 26.3 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | zeit/next.js canary  | tomevans18/next.js with-docker-example-public-folder | Change |
| - | - | - | - |
| _app.js gzip | 1.26 kB | 1.26 kB | ✓ |
| _error.js gzip | 3.29 kB | 3.29 kB | ✓ |
| hooks.js gzip | 880 B | 880 B | ✓ |
| index.js gzip | 222 B | 222 B | ✓ |
| link.js gzip | 2.06 kB | 2.06 kB | ✓ |
| routerDirect.js gzip | 280 B | 280 B | ✓ |
| withRouter.js gzip | 278 B | 278 B | ✓ |
| Overall change | 8.26 kB | 8.26 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages Modern</strong></summary>

|  | zeit/next.js canary  | tomevans18/next.js with-docker-example-public-folder | Change |
| - | - | - | - |
| _app.module.js gzip | 599 B | 599 B | ✓ |
| _error.module.js gzip | 2.09 kB | 2.09 kB | ✓ |
| hooks.module.js gzip | 383 B | 383 B | ✓ |
| index.module.js gzip | 223 B | 223 B | ✓ |
| link.module.js gzip | 1.52 kB | 1.52 kB | ✓ |
| routerDirect..dule.js gzip | 279 B | 279 B | ✓ |
| withRouter.m..dule.js gzip | 278 B | 278 B | ✓ |
| Overall change | 5.37 kB | 5.37 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | zeit/next.js canary  | tomevans18/next.js with-docker-example-public-folder | Change |
| - | - | - | - |
| _buildManifest.js gzip | 61 B | 61 B | ✓ |
| _buildManife..dule.js gzip | 61 B | 61 B | ✓ |
| Overall change | 122 B | 122 B | ✓ |

</details>

<details>
<summary><strong>Serverless bundles</strong></summary>

|  | zeit/next.js canary  | tomevans18/next.js with-docker-example-public-folder | Change |
| - | - | - | - |
| _error.js | 850 kB | 850 kB | ✓ |
| 404.html | 4.18 kB | 4.18 kB | ✓ |
| hooks.html | 3.81 kB | 3.81 kB | ✓ |
| index.js | 851 kB | 851 kB | ✓ |
| link.js | 888 kB | 888 kB | ✓ |
| routerDirect.js | 880 kB | 880 kB | ✓ |
| withRouter.js | 880 kB | 880 kB | ✓ |
| Overall change | 4.36 MB | 4.36 MB | ✓ |

</details>

</details>

File: examples/with-docker/next.config.js [hunk 1]
```diff
@@ -0,0 +1,3 @@
+module.exports = {
+  /* config options here */
+}
```
Comment (lfades) [inline examples/with-docker/next.config.js:3]: Is this file required?

Comment (tomevans18) [reply examples/with-docker/next.config.js:3]: It is needed for the multistage Dockerfile to build.
I felt this was wise to include as someone pulling the example would expect the Dockerfile to simply work with no modification required

**Comment by ijjk** [comment]
# Stats from current PR

<details>
<summary><strong>Default Server Mode</strong> (Decrease detected ✓)</summary>

<br/>

<details>
<summary><strong>General</strong></summary>

|  | zeit/next.js canary  | tomevans18/next.js with-docker-example-public-folder | Change |
| - | - | - | - |
| buildDuration | 12.5s | 12.7s | ⚠️ +201ms |
| nodeModulesSize | 58.7 MB | 58.7 MB | ✓ |

</details>

<details>
<summary><strong>Page Load Tests</strong> Overall decrease ⚠️</summary>

|  | zeit/next.js canary  | tomevans18/next.js with-docker-example-public-folder | Change |
| - | - | - | - |
| / failed reqs | 0 | 0 | ✓ |
| / total time (seconds) | 2.024 | 2.037 | ⚠️ +0.01 |
| / avg req/sec | 1235.11 | 1227.35 | -7.76 |
| /error-in-render failed reqs | 0 | 0 | ✓ |
| /error-in-render total time (seconds) | 1.236 | 1.277 | ⚠️ +0.04 |
| /error-in-render avg req/sec | 2022.08 | 1957.06 | -65.02 |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | zeit/next.js canary  | tomevans18/next.js with-docker-example-public-folder | Change |
| - | - | - | - |
| main-HASH.js gzip | 6.68 kB | 6.68 kB | ✓ |
| webpack-HASH.js gzip | 746 B | 746 B | ✓ |
| de003c3a9d30..e49d.js gzip | 10.4 kB | 10.4 kB | ✓ |
| framework.HASH.js gzip | 39.1 kB | 39.1 kB | ✓ |
| Overall change | 57 kB | 57 kB | ✓ |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons) Modern</strong></summary>

|  | zeit/next.js canary  | tomevans18/next.js with-docker-example-public-folder | Change |
| - | - | - | - |
| main-HASH.module.js gzip | 5.72 kB | 5.72 kB | ✓ |
| webpack-HASH..dule.js gzip | 746 B | 746 B | ✓ |
| de003c3a9d30..dule.js gzip | 6.85 kB | 6.85 kB | ✓ |
| framework.HA..dule.js gzip | 39.1 kB | 39.1 kB | ✓ |
| Overall change | 52.5 kB | 52.5 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | zeit/next.js canary  | tomevans18/next.js with-docker-example-public-folder | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 26.3 kB | 26.3 kB | ✓ |
| Overall change | 26.3 kB | 26.3 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | zeit/next.js canary  | tomevans18/next.js with-docker-example-public-folder | Change |
| - | - | - | - |
| _app.js gzip | 1.26 kB | 1.26 kB | ✓ |
| _error.js gzip | 3.29 kB | 3.29 kB | ✓ |
| hooks.js gzip | 880 B | 880 B | ✓ |
| index.js gzip | 222 B | 222 B | ✓ |
| link.js gzip | 2.06 kB | 2.06 kB | ✓ |
| routerDirect.js gzip | 280 B | 280 B | ✓ |
| withRouter.js gzip | 278 B | 278 B | ✓ |
| Overall change | 8.26 kB | 8.26 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages Modern</strong></summary>

|  | zeit/next.js canary  | tomevans18/next.js with-docker-example-public-folder | Change |
| - | - | - | - |
| _app.module.js gzip | 599 B | 599 B | ✓ |
| _error.module.js gzip | 2.09 kB | 2.09 kB | ✓ |
| hooks.module.js gzip | 383 B | 383 B | ✓ |
| index.module.js gzip | 223 B | 223 B | ✓ |
| link.module.js gzip | 1.52 kB | 1.52 kB | ✓ |
| routerDirect..dule.js gzip | 279 B | 279 B | ✓ |
| withRouter.m..dule.js gzip | 278 B | 278 B | ✓ |
| Overall change | 5.37 kB | 5.37 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | zeit/next.js canary  | tomevans18/next.js with-docker-example-public-folder | Change |
| - | - | - | - |
| _buildManifest.js gzip | 61 B | 61 B | ✓ |
| _buildManife..dule.js gzip | 61 B | 61 B | ✓ |
| Overall change | 122 B | 122 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | zeit/next.js canary  | tomevans18/next.js with-docker-example-public-folder | Change |
| - | - | - | - |
| index.html gzip | 926 B | 926 B | ✓ |
| link.html gzip | 936 B | 936 B | ✓ |
| withRouter.html gzip | 923 B | 923 B | ✓ |
| Overall change | 2.79 kB | 2.79 kB | ✓ |

</details>

</details>
<hr/>
<details>
<summary><strong>Serverless Mode</strong></summary>

<br/>

<details>
<summary><strong>General</strong></summary>

|  | zeit/next.js canary  | tomevans18/next.js with-docker-example-public-folder | Change |
| - | - | - | - |
| buildDuration | 13.7s | 13.4s | -299ms |
| nodeModulesSize | 58.7 MB | 58.7 MB | ✓ |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | zeit/next.js canary  | tomevans18/next.js with-docker-example-public-folder | Change |
| - | - | - | - |
| main-HASH.js gzip | 6.68 kB | 6.68 kB | ✓ |
| webpack-HASH.js gzip | 746 B | 746 B | ✓ |
| de003c3a9d30..e49d.js gzip | 10.4 kB | 10.4 kB | ✓ |
| framework.HASH.js gzip | 39.1 kB | 39.1 kB | ✓ |
| Overall change | 57 kB | 57 kB | ✓ |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons) Modern</strong></summary>

|  | zeit/next.js canary  | tomevans18/next.js with-docker-example-public-folder | Change |
| - | - | - | - |
| main-HASH.module.js gzip | 5.72 kB | 5.72 kB | ✓ |
| webpack-HASH..dule.js gzip | 746 B | 746 B | ✓ |
| de003c3a9d30..dule.js gzip | 6.85 kB | 6.85 kB | ✓ |
| framework.HA..dule.js gzip | 39.1 kB | 39.1 kB | ✓ |
| Overall change | 52.5 kB | 52.5 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | zeit/next.js canary  | tomevans18/next.js with-docker-example-public-folder | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 26.3 kB | 26.3 kB | ✓ |
| Overall change | 26.3 kB | 26.3 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | zeit/next.js canary  | tomevans18/next.js with-docker-example-public-folder | Change |
| - | - | - | - |
| _app.js gzip | 1.26 kB | 1.26 kB | ✓ |
| _error.js gzip | 3.29 kB | 3.29 kB | ✓ |
| hooks.js gzip | 880 B | 880 B | ✓ |
| index.js gzip | 222 B | 222 B | ✓ |
| link.js gzip | 2.06 kB | 2.06 kB | ✓ |
| routerDirect.js gzip | 280 B | 280 B | ✓ |
| withRouter.js gzip | 278 B | 278 B | ✓ |
| Overall change | 8.26 kB | 8.26 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages Modern</strong></summary>

|  | zeit/next.js canary  | tomevans18/next.js with-docker-example-public-folder | Change |
| - | - | - | - |
| _app.module.js gzip | 599 B | 599 B | ✓ |
| _error.module.js gzip | 2.09 kB | 2.09 kB | ✓ |
| hooks.module.js gzip | 383 B | 383 B | ✓ |
| index.module.js gzip | 223 B | 223 B | ✓ |
| link.module.js gzip | 1.52 kB | 1.52 kB | ✓ |
| routerDirect..dule.js gzip | 279 B | 279 B | ✓ |
| withRouter.m..dule.js gzip | 278 B | 278 B | ✓ |
| Overall change | 5.37 kB | 5.37 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | zeit/next.js canary  | tomevans18/next.js with-docker-example-public-folder | Change |
| - | - | - | - |
| _buildManifest.js gzip | 61 B | 61 B | ✓ |
| _buildManife..dule.js gzip | 61 B | 61 B | ✓ |
| Overall change | 122 B | 122 B | ✓ |

</details>

<details>
<summary><strong>Serverless bundles</strong></summary>

|  | zeit/next.js canary  | tomevans18/next.js with-docker-example-public-folder | Change |
| - | - | - | - |
| _error.js | 850 kB | 850 kB | ✓ |
| 404.html | 4.18 kB | 4.18 kB | ✓ |
| hooks.html | 3.81 kB | 3.81 kB | ✓ |
| index.js | 851 kB | 851 kB | ✓ |
| link.js | 888 kB | 888 kB | ✓ |
| routerDirect.js | 880 kB | 880 kB | ✓ |
| withRouter.js | 880 kB | 880 kB | ✓ |
| Overall change | 4.36 MB | 4.36 MB | ✓ |

</details>

</details>

File: examples/with-docker/next.config.js (hunk 1 above)
Comment (timneutkens) [reply examples/with-docker/next.config.js:3]: Indeed not required

**Review by timneutkens — APPROVED** [review]

File: examples/with-docker/next.config.js (hunk 1 above)
Comment (tomevans18) [reply examples/with-docker/next.config.js]: When I say its required, this is because the Dockerfile has always contained the following line - https://github.com/zeit/next.js/pull/12879/files#diff-c7b5f2152f889efdbdf9e7615c2cb65bL14

This means `docker build -t next-app -f ./Dockerfile.multistage .` as per the readme notes will fail, as the file `next.config.js` did not previously exist. It is possibly wiser to comment out this line in the Dockerfile as `next.config.js` is not required to run Next.js but people will understand they may need to comment this line back in.

Comment (timneutkens) [reply examples/with-docker/next.config.js]: Sounds fine!

**Review by Timer — APPROVED** [review]
